libsolidity/inlineasm/AsmAnalysis.cpp

```diff
@@ -486,4 +486,11 @@ void AsmAnalyzer::warnOnFutureInstruction(solidity::Instruction _instr, SourceLo
 			+ "\" instruction is only available after " +
 			"the Metropolis hard fork. Before that it acts as an invalid instruction."
 		);
+
+	if (_instr == solidity::Instruction::JUMP || _instr == solidity::Instruction::JUMPI)
+		m_errorReporter.warning(
+			_location,
+			"Using jumps requires stack adjustments when working with variables. "
```

Comment:
"requires" sounds too strong, jumpi can be used without stack adjustments.
Perhaps we should phase the whole thing in a different way and also warn on labels. Something like `labels and jump instructions are low-level EVM features that can lead to incorrect stack access. Because of that, they are discouraged, please consider using "switch" or "for" statements."